.pre-commit-config.yaml

```diff
@@ -47,3 +47,8 @@ repos:
           - python
         exclude: vendored
         args: [--config, python/setup.cfg]
+  - repo: https://github.com/MarcoGorelli/cython-lint
+    rev: v0.12.4
+    hooks:
+      - id: cython-lint
+        args: [--max-line-length=120]
```

Comment:
nit: flake8 uses 88 as line length, should this match or are the reasons to increase it for cython files?

Reply:
Thanks for your review

It was just to keep the diff down until there was some agreement on whether you wanted this

If you're on board with it, I can go ahead and reduce it to 88

Reply:
Hi @assignUser - there are some doctests where the result is longer than 88 lines, e.g.

https://github.com/apache/arrow/blob/1ab1f6f159ebf4df1216f28288207a357bbb4017/python/pyarrow/table.pxi#L3843-L3849

` # doctest: +NORMALIZE_WHITESPACE` won't work here (as far as I can tell) because it's a long string without whitespace,
so inserting a newline will make doctest fail

Is it OK to `doctest: +SKIP` this one?

Reply:
Sounds good to me but I am not a python maintainer so I will defer to @jorisvandenbossche :)

Reply:
I would suggest making the docstring example data smaller, like so:

```python
>>> import pyarrow as pa
>>> n_legs = pa.array([2, 4, 100])
>>> names = ["n_legs", "animals"]
>>> batch = pa.record_batch([n_legs, animals], names=names)
>>> batch.to_pandas()
Construct a Table   n_legs    animals
0       2   Flamingo
1       4      Horse
2     100  Centipede

Construct a Table from a RecordBatch:

>>> pa.Table.from_batches([batch])
pyarrow.Table
n_legs: int64
animals: string
----
n_legs: [[2,4,100]]
animals: [["Flamingo","Horse","Centipede"]]

Construct a Table from a sequence of RecordBatches:

>>> pa.Table.from_batches([batch, batch])
pyarrow.Table
n_legs: int64
animals: string
----
n_legs: [[2,4,100],[2,4,100]]
animals: [["Flamingo","Horse","Centipede"],["Flamingo","Horse","Centipede"]]
```
I think that should get under the limit of 88.

Reply:
thanks for your reviews

for now I've put the `--no-pycodestyle` flag, which only does the linting checks (e.g. unused variables/imports/dangerous patterns, rather than style nitpicks) - is that OK?

There's hundreds of cases which would need fixing to get under the 88 limit, which at the moment I'm a bit too busy for I'm afraid

Else I'll update with the 88 limit when I get a chance, no worries

Reply:
I think that is fine 👍 
I would suggest to open an issue to update the 88 limit as a follow up so that it can be tracked and we do not forget to do that at some point.